articles/active-directory/develop/tutorial-v2-windows-uwp.md

```diff
@@ -216,43 +222,31 @@ Eventually, the `AcquireTokenSilent` method fails. Reasons for failure include a
 
 * Your application presents a visual indication to users that they need to sign in. Then they can select the right time to sign in. The application can retry `AcquireTokenSilent` later. Use this approach when users can use other application functionality without disruption. An example is when offline content is available in the application. In this case, users can decide when they want to sign in. The application can retry `AcquireTokenSilent` after the network was temporarily unavailable.
 
-### Call Microsoft Graph API by using the token you just obtained
+### Instantiate Microsoft Graph Service Client by using the token you just obtained
 
 Add the following new method to *MainPage.xaml.cs*:
 
    ```csharp
-   /// <summary>
-   /// Perform an HTTP GET request to a URL using an HTTP Authorization header
-   /// </summary>
-   /// <param name="url">The URL</param>
-   /// <param name="token">The token</param>
-   /// <returns>String containing the results of the GET operation</returns>
-   public async Task<string> GetHttpContentWithToken(string url, string token)
-   {
-       var httpClient = new System.Net.Http.HttpClient();
-       System.Net.Http.HttpResponseMessage response;
-       try
-       {
-           var request = new System.Net.Http.HttpRequestMessage(System.Net.Http.HttpMethod.Get, url);
-           // Add the token in Authorization header
-           request.Headers.Authorization =
-             new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", token);
-           response = await httpClient.SendAsync(request);
-           var content = await response.Content.ReadAsStringAsync();
-           return content;
-       }
-       catch (Exception ex)
+         /// <summary>
+        /// Instantiating GraphServiceClient using the access token.
+        /// </summary>
+        /// <param name="token">The token</param>
+        /// <returns>GraphServiceClient</returns>
+        private GraphServiceClient GetGraphServiceClient(string token)
         {
-           return ex.ToString();
-       }
-    }
+            GraphServiceClient graphServiceClient = new GraphServiceClient(new DelegateAuthenticationProvider((requestMessage) =>
```

Comment:
We should NOT encourage customers to do that. this is very fragile, and had a big performance hit.